dev/deploy-site.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+set -e
+
+usage () {
+    echo "Usage: $0 [-r <git_remote>]"
+    echo "  -r      Specify Git remote name. Defaults to \"apache\""
+    exit 1
+}
+
+# Default repository remote name
+remote="apache"
+
+while getopts "r:" opt; do
+  case "${opt}" in
+    r)
+      remote="${OPTARG}"
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+if [[ -n $1 ]]; then
+  >&2 echo "Error: unrecognized argument '$1'"
+  usage
+fi
+
+scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+projectdir="$(dirname "$scriptdir")"
+
+pushd .
+trap "popd" EXIT
+
+# If the following commands stop working, please check this ASF INFRA
+# page for updates: https://infra.apache.org/asfyaml-mkdocs.html
+cd ${projectdir}/site
+rm -rf site
+mkdir site
+cp ../.asf.yaml site/
+mkdocs gh-deploy --dirty -r ${remote}
```

Comment:
What does the `--dirty` flag do? Do we need it every time? Might be worth putting a comment on why we need it (for example, in case we don't need it in the future).

Reply:
@rdblue provided an explanation here: https://github.com/apache/iceberg/pull/2998#discussion_r693517612
I will update the script to include an explanation.